### PR TITLE
Add PDF weight helper tool (80x)

### PR DESCRIPTION
**Pull request by bendavid**
To be used at analysis level to apply the mc2hessian procedure on the fly for analysis needing hessian-type uncertainties from existing samples with mc replica-like weights.

### CODE REVIEW

**Comment by davidlange6** [comment]
@cmsbuild, please test

File: PhysicsTools/HepMCCandAlgos/plugins/PDFWeightsTest.cc [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+#include "FWCore/Utilities/interface/EDMException.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "TTree.h"
+#include <iostream>
+using namespace std;
+using namespace edm;
+using namespace HepMC;
+
+class PDFWeightsTest : public EDAnalyzer {
```
Comment (davidlange6) [inline PhysicsTools/HepMCCandAlgos/plugins/PDFWeightsTest.cc]: @bendavid - could you change this to inherit from a non-legacy module type. Thanks!

**Comment by davidlange6** [comment]
@bendavid - ping

**Comment by bendavid** [comment]
Yup, haven't forgotten, will get back to this soon.

**Comment by davidlange6** [comment]
Thanks!

**Comment by bendavid** [comment]
Sorry this one got messed up and has been replaced by https://github.com/cms-sw/cmssw/pull/13019

**Pull request closed without merging**